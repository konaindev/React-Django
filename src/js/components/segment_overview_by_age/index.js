import React, { Component } from "react";
import cn from "classnames";
import PropTypes from "prop-types";

import { formatNumber } from "../../utils/formatters";
import MarketSizeByIncome from "../market_size_by_income";
import "./segment_overview_by_age.css";

export class SegmentOverviewByAge extends Component {
  static propTypes = {
    age_group: PropTypes.string.isRequired,
    income_groups: PropTypes.array.isRequired,
    segment_number: PropTypes.number.isRequired,
    segment_population: PropTypes.number.isRequired,
    total_population: PropTypes.number.isRequired
  };

  render() {
    const {
      age_group,
      income_groups,
      segment_number,
      segment_population,
      total_population
    } = this.props;
    return (
      <div className="segment-overview-by-age">
        <div className="segment-overview-by-age__heading">
          <div className="segment-overview-by-age__heading-left">
            Segment {segment_number} | Ages {age_group}
          </div>
          <div className="segment-overview-by-age__heading-right">
            Est. Segment Population {formatNumber(segment_population)}
            <span className="segment-overview-by-age__total-pop">
              {` / ${formatNumber(total_population)}`}
            </span>
          </div>
        </div>
        <div className="segment-overview-by-age__content">
          {income_groups.map((item, index) => (
            <MarketSizeByIncome
              key={index}
              {...item}
              segment_population={segment_population}
            />
          ))}
        </div>
      </div>
    );
  }
}

export default SegmentOverviewByAge;
