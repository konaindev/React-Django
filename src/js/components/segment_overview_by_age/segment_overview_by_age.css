.segment-overview-by-age {
  border: 1px solid #161C29;
  border-radius: 6px;
  background-color: #1A202E;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.15);
  padding: 40px 60px;
}

.segment-overview-by-age__heading {
  padding-bottom: 40px;
  color: #ffffff;
  border-bottom: 1px solid #2c3646;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.segment-overview-by-age__heading-left {
  font-size: 1rem;
}

.segment-overview-by-age__heading-right {
  font-size: 0.875rem;
}

.segment-overview-by-age__total-pop {
  color: #566278;
}

.segment-overview-by-age__content {
  display: flex;
  margin: 0 -48px;
}
