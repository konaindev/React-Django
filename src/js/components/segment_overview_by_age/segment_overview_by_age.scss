@import "css/base.scss";

// TODO FIXME use the contents of variables.scss for colors etc.

.segment-overview-by-age {
    padding: 40px 60px;
    margin: $panel-spacing;
}

.segment-overview-by-age__heading {
    padding-bottom: 40px;
    color: #ffffff;
    border-bottom: 1px solid #2c3646;
    display: flex;
    align-items: center;
    justify-content: space-between;
}

.segment-overview-by-age__heading-left {
    font-size: 1rem;
}

.segment-overview-by-age__heading-right {
    font-size: 0.875rem;
}

.segment-overview-by-age__total-pop {
    color: #566278;
}

.segment-overview-by-age__content {
    display: flex;
    margin: 0 -48px;
}
