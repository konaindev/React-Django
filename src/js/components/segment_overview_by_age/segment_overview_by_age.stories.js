import React from "react";

import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";
import { linkTo } from "@storybook/addon-links";
import SegmentOverviewByAge from "./index";

export const props = {
  age_group: "18-24",
  segment_population: 10368,
  total_population: 120448,
  segment_number: 1,
  income_groups: [
    {
      income: "50000.00",
      group_population: 3452,
      home_owners: {
        total: 231,
        family: 114,
        nonfamily: 117
      },
      renters: {
        total: 3311,
        family: 517,
        nonfamily: 2794
      },
      active_populations: ["renters.nonfamily"],
      market_size: 2794
    },
    {
      income: "75000.00",
      group_population: 3188,
      home_owners: {
        total: 249,
        family: 123,
        nonfamily: 126
      },
      renters: {
        total: 2940,
        family: 459,
        nonfamily: 2481
      },
      market_size: 2481,
      active_populations: ["renters.nonfamily"]
    },
    {
      income: "100000.00",
      group_population: 1587,
      home_owners: {
        total: 164,
        family: 81,
        nonfamily: 83
      },
      renters: {
        total: 1423,
        family: 83,
        nonfamily: 1201
      },
      market_size: 1201,
      active_populations: ["renters.nonfamily"]
    }
  ]
};

storiesOf("SegmentOverviewByAge", module).add("default", () => (
  <SegmentOverviewByAge {...props} />
));
