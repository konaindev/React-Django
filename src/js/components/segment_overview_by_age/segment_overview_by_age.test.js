import renderer from "react-test-renderer";
import SegmentOverviewByAge from "./index";
import { props } from "./segment_overview_by_age.stories";

describe("SegmentOverviewByAge", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<SegmentOverviewByAge {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
