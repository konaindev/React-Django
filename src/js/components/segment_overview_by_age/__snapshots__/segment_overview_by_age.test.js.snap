// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentOverviewByAge renders correctly 1`] = `
"<Panel className=\\"segment-overview-by-age\\" component=\\"div\\" size=\\"default\\">
  <div className=\\"segment-overview-by-age__heading\\">
    <div className=\\"segment-overview-by-age__heading-left\\">
      Segment 
      1
       | Ages 
      18-24
    </div>
    <div className=\\"segment-overview-by-age__heading-right\\">
      Est. Segment Population 
      10,368
      <span className=\\"segment-overview-by-age__total-pop\\">
         / 120,448
      </span>
      <Connect(InfoTooltipBase) transKey=\\"est_segment_population\\" />
    </div>
  </div>
  <div className=\\"segment-overview-by-age__content\\">
    <MarketSizeByIncome income=\\"50000.00\\" group_population={3452} home_owners={{...}} renters={{...}} active_populations={{...}} market_size={2794} segment_population={10368} />
    <MarketSizeByIncome income=\\"75000.00\\" group_population={3188} home_owners={{...}} renters={{...}} market_size={2481} active_populations={{...}} segment_population={10368} />
    <MarketSizeByIncome income=\\"100000.00\\" group_population={1587} home_owners={{...}} renters={{...}} market_size={1201} active_populations={{...}} segment_population={10368} />
  </div>
</Panel>"
`;
